app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours/OpeningHoursTagParser.kt

```diff
@@ -0,0 +1,463 @@
+        return returned
+    }
+
+    // StreetComplete supports solely date changing based on month
+    // without any support for any other data ranges
+    // function may return identical or modified object or null
+    // null or modified object indicates that original object was not representable in SC
+    private fun reduceDateRangeToFullMonths(dateRange: DateRange): DateRange? {
+        for (date in arrayOf(dateRange.startDate, dateRange.endDate).filterNotNull()) {
+            if (date.isOpenEnded) {
+                return null //TODO: it may be supported by StreetComplete
+            }
+            if (date.weekDayOffset != null) {
+                return null
+            }
+            if (date.dayOffset != 0) {
+                return null
+            }
+        }
+        val newDateRange = DateRange()
+
+        val startDate = DateWithOffset()
+        startDate.month = dateRange.startDate.month
+        newDateRange.startDate = startDate
+
+        val endDate = dateRange.endDate
+        if (endDate != null) {
+            // range with just single month will have endDate unset
+            val newEndDate = DateWithOffset()
+            newEndDate.month = endDate.month
+            newDateRange.endDate = newEndDate
+        }
+        return newDateRange
+    }
+
+    // StreetComplete has no support for times like "from sunrise to sunset"
+    // this function throws away any info over "from hour X to hour Y"
+    // function may return identical or modified object or null
+    // null or modified object indicates that original object was not representable in SC
+    private fun reduceTimeRangeToSimpleTime(timeSpan: TimeSpan): TimeSpan? {
+        val simplifiedTimespan = TimeSpan()
+        if (timeSpan.startEvent != null) {
+            return null
+        }
+        if (timeSpan.endEvent != null) {
+            return null
+        }
+        val startInMinutesSinceMidnight = timeSpan.start
+        if (startInMinutesSinceMidnight < 0) {
+            return null
+        }
+        if (startInMinutesSinceMidnight > 24 * 60) {
+            return null
+        }
+        simplifiedTimespan.start = startInMinutesSinceMidnight
+        val endInMinutesSinceMidnight = timeSpan.end
+        if (endInMinutesSinceMidnight < 0) {
+            return null
+        }
+        simplifiedTimespan.end = endInMinutesSinceMidnight
+        return simplifiedTimespan
+    }
+
+    private fun emptyRule(): Rule {
+        // workaround needed to construct empty Rule object
+        // proposal to allow creation of Rule objects is at
+        // https://github.com/simonpoole/OpeningHoursParser/pull/24
+        val input = ByteArrayInputStream("".toByteArray())
+        val parser = OpeningHoursParser(input)
+        try {
+            val rules = parser.rules(true)
+            return rules[0]
+        } catch (e: ParseException) {
+            e.printStackTrace()
+            throw RuntimeException()
+        }
```

Comment:
Debugging code leftovers

Reply:
Not exactly. It catches ParseException and rethrows RuntimeException.

ParseException must be caught at some level and RuntimeException is allowed to not be caught.

And given that this will not happen except situation of library making some incompatible changes I think that `RuntimeException` is OK here.

Reply:
In Kotlin, there are no checked exceptions. You sure you need to catch it?

Anyway: Ok